setup.py

```diff
@@ -11,35 +11,37 @@
 
 exec(compile(version_file_contents, "pymbolic/version.py", "exec"), ver_dic)
 
-setup(name="pymbolic",
-      version=ver_dic["VERSION_TEXT"],
-      description="A package for symbolic computation",
-      long_description=open("README.rst").read(),
-      classifiers=[
-          "Development Status :: 4 - Beta",
-          "Intended Audience :: Developers",
-          "Intended Audience :: Other Audience",
-          "Intended Audience :: Science/Research",
-          "License :: OSI Approved :: MIT License",
-          "Natural Language :: English",
-          "Programming Language :: Python",
-          "Programming Language :: Python :: 3",
-          "Topic :: Scientific/Engineering",
-          "Topic :: Scientific/Engineering :: Mathematics",
-          "Topic :: Software Development :: Libraries",
-          "Topic :: Utilities",
-          ],
-      author="Andreas Kloeckner",
-      author_email="inform@tiker.net",
-      license="MIT",
-      url="http://mathema.tician.de/software/pymbolic",
-
-      packages=find_packages(),
-      python_requires="~=3.8",
-      install_requires=[
-          "pytools>=2022.1.14",
-          ],
-      extras_require={
-          "test": ["pytest>=2.3"],
-          },
-      )
+setup(
+    name="pymbolic",
+    version=ver_dic["VERSION_TEXT"],
+    description="A package for symbolic computation",
+    long_description=open("README.rst").read(),
+    classifiers=[
+        "Development Status :: 4 - Beta",
+        "Intended Audience :: Developers",
+        "Intended Audience :: Other Audience",
+        "Intended Audience :: Science/Research",
+        "License :: OSI Approved :: MIT License",
+        "Natural Language :: English",
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 3",
+        "Topic :: Scientific/Engineering",
+        "Topic :: Scientific/Engineering :: Mathematics",
+        "Topic :: Software Development :: Libraries",
+        "Topic :: Utilities",
+    ],
+    author="Andreas Kloeckner",
+    author_email="inform@tiker.net",
+    license="MIT",
+    url="http://mathema.tician.de/software/pymbolic",
+    packages=find_packages(),
+    python_requires="~=3.8",
+    install_requires=[
+        "pytools>=2022.1.14",
+        "attrs",
+        "immutables",
```

Comment:
`immutables` is only needed to create a hashable mapping for handling `expr(kwarg=value)`. This could be a soft dependency.

Reply:
Decided to stick with `immutables` as a hard dependency.